cmake/bazel_to_cmake.py

```diff
@@ -0,0 +1,282 @@
+        elif re.search(r':do_not_want_O3$', key):
+            # Ruy is a specialist library: we always want code to be compiled
+            # with -O3 unless the build type is Debug or the compiler does not
+            # support that flag syntax.
+            condition = '(CMAKE_BUILD_TYPE STREQUAL Debug) OR MSVC'
+        elif re.search(r':x86_64_and_not_msvc$', key):
+            condition = '(CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64 OR CMAKE_SYSTEM_PROCESSOR STREQUAL amd64) AND NOT MSVC'
+        elif re.search(r':windows_msvc$', key):
+            condition = 'MSVC'
+        elif re.search(r':ruy_profiler$', key):
+            condition = '${RUY_PROFILER}'
+        else:
+            raise ValueError(f'Unhandled key in select: {key}')
+
+        print(f'{new_if_branch_keyword}({condition})')
+        set_cmake_list(select_name, dict[key], '  ')
+        new_if_branch_keyword = 'elseif'
+
+    print('else()')
+    set_cmake_list(select_name, default_value, '  ')
+
+    print('endif()\n')
+
+
+def trim_multiple_ruy_prefixes(name):
+    return re.sub(r'(ruy_)+ruy', 'ruy', name)
+
+def get_cmake_local_target_name(name):
+    global package_prefix
+    return trim_multiple_ruy_prefixes(f'ruy_{package_prefix}_{name}')
+
+
+def get_cmake_dep_target_name(name):
+    if name in external_targets:
+        return name
+    if name.startswith('$'):
+        # Happens for deps that are the result of expanding a select() that we
+        # have compiled to expanding a variable.
+        return name
+    if name.startswith('//'):
+        after_last_slash = name.split('/')[-1]
+        if not ':' in after_last_slash:
+            name = f'{name}:{after_last_slash}'
+        raw=name[2:].replace('/', '_').replace(':', '_')
+        return trim_multiple_ruy_prefixes(raw)
+    if name.startswith(':'):
+        name = name[1:]
+    return get_cmake_local_target_name(name)
+
+
+#
+# Functions implementing BUILD functions
+#
+
+
+def package(**kwargs):
+    pass
+
+
+def exports_files(*args):
+    pass
+
+
+def load(filename, *args):
+    if filename.startswith('@'):
+        return
+    elif filename.startswith(':'):
+        filename = os.path.join(bazel_package_dir, filename[1:])
+    elif filename.startswith('//'):
+        split = filename[2:].split(':')
+        filename = os.path.join(bazel_workspace_dir, split[0], split[1])
+
+    src_file_content = open(filename).read()
+    processed_file_content = preprocess_input_text(src_file_content)
+    exec(processed_file_content, globals(), globals())
+
+
+def config_setting(**kwargs):
+    # Nothing to do since our implementation of select() is based on parsing
+    # the names of config_settings, not looking deep into their actual
+    # implementation.
+    pass
+
+
+def filegroup(**kwargs):
+    pass
+
+
+def config_setting_group(**kwargs):
+    # See config_setting.
+    pass
+
+
+def bzl_library(**kwargs):
+    pass
+
+
+select_index = 0
+select_cache = {}
+
+
+def select(dict):
```

Comment:
```suggestion
def select(select_dict):
```

`dict` is a builtin/keyword. Probably best not to use.

Reply:
thanks!